app/src/main/java/org/mozilla/reference/browser/NotificationManager.kt

```diff
@@ -0,0 +1,126 @@
+import mozilla.components.concept.sync.Device
+import mozilla.components.concept.sync.TabData
+import mozilla.components.support.base.log.logger.Logger
+
+/**
+ * Manages notification channels and allows displaying different types of notifications.
+ */
+class NotificationManager(private val context: Context) {
+    companion object {
+        const val RECEIVE_TABS_TAG = "ReceivedTabs"
+        const val RECEIVE_TABS_CHANNEL_ID = "ReceivedTabsChannel"
+    }
+
+    init {
+        // Create the notification channels we are going to use, but only on API 26+ because the NotificationChannel
+        // class is new and not in the support library.
+        if (SDK_INT >= Build.VERSION_CODES.O) {
+            createNotificationChannel(
+                RECEIVE_TABS_CHANNEL_ID,
+                // Pick 'high' because this is a user-triggered action that is expected to be part of a continuity flow.
+                // That is, user is expected to be waiting for this notification on their device; make it obvious.
+                AndroidNotificationManager.IMPORTANCE_HIGH,
+                // Name and description are shown in the 'app notifications' settings for the app.
+                context.getString(R.string.fxa_received_tab_channel_name),
+                context.getString(R.string.fxa_received_tab_channel_description)
+            )
+        }
+    }
+
+    private val logger = Logger("NotificationManager")
+    fun showReceivedTabs(device: Device?, tabs: List<TabData>) {
+        // In the future, experiment with displaying multiple tabs from the same device as as Notification Groups.
+        // For now, a single notification per tab received will suffice.
+        logger.debug("Showing ${tabs.size} tab(s) received from deviceID=${device?.id}")
+        tabs.forEach { tab ->
+            val intent = Intent(Intent.ACTION_VIEW, Uri.parse(tab.url)).apply {
+                flags = Intent.FLAG_ACTIVITY_NEW_TASK
+            }
+            val pendingIntent: PendingIntent =
+                PendingIntent.getActivity(context, 0, intent, PendingIntent.FLAG_ONE_SHOT)
+
+            val builder = NotificationCompat.Builder(context, RECEIVE_TABS_CHANNEL_ID)
+                .setSmallIcon(R.drawable.ic_status_logo)
+                .setSendTabTitle(context, device, tab)
+                .setWhen(System.currentTimeMillis())
+                .setContentText(tab.url)
+                .setContentIntent(pendingIntent)
+                .setAutoCancel(true)
+                .setPriority(NotificationCompat.PRIORITY_HIGH)
+                .setDefaults(Notification.DEFAULT_VIBRATE or Notification.DEFAULT_SOUND)
+
+            if (SDK_INT >= Build.VERSION_CODES.M) {
+                builder.setCategory(Notification.CATEGORY_REMINDER)
+            }
+
+            // Pick a random ID for this notification so that different tabs do not clash.
+            @SuppressWarnings("MagicNumber")
+            val notificationId = (Math.random() * 100).toInt()
```

Comment:
Should we use [NotificationIds](https://github.com/mozilla-mobile/android-components/blob/6e6311ea44f8f24ec6339d89a5a462a779cee8d0/components/support/base/src/main/java/mozilla/components/support/base/ids/NotificationIds.kt#L24)?

Reply:
Yes please. :)

Reply:
Wouldn't this give me the same id all the time? I don't have different tags, but I need new IDs generated for each tab sent.

Reply:
Ah, yeah, that is tricky. You could still create a different tag based on the URL every time. But that is also wasteful.

Anyhow, using random will just randomly replace other notifications and is not a good choice.

Reply:
I took the NotificationManager [from Fenix](https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/java/org/mozilla/fenix/components/NotificationManager.kt), so I wonder how that's working 🤔 .

I guess I should store the IDs in a map to avoid collisions.

Reply:
Since we're using a unique tag for send tab, I added a simple notification ID counter that would always increment up. That should avoid notification ID collisions.